pkg/reflectutil/reflectutil_test.go

```diff
@@ -0,0 +1,154 @@
+		IntPtr2:  nil,
+		IntPtr3:  pointer.Int(300),
+		IntPtr4:  pointer.Int(0),
+		StrPtr:   pointer.String("olleh"),
+		StrPtr2:  nil,
+		StrStrMap: map[string]string{
+			"b": "beer",
+			"c": "",
+			"d": "daiquiri",
+		},
+		StrSlice: []string{"gamma", "delta"},
+		StrStrMap2: map[string]string{
+			"a": "america",
+			"b": "brazil",
+		},
+		StrArray: [2]string{"california", "colorado"},
+		Struct: FooChild{
+			Int:     -43,
+			Str:     "ruojnob",
+			IP:      net.ParseIP("192.168.11.1"),
+			IPSlice: []net.IP{net.ParseIP("10.0.0.3")},
+			StrStrPtrMap: map[string]*string{
+				"a": nil,
+				"b": pointer.String("bacardi"),
+				"c": pointer.String("cinzano"),
+				"d": pointer.String(""),
+			},
+		},
+	}
+
+	expected := &Foo{
+		BoolPtr:  pointer.Bool(false),     // overridden
+		BoolPtr2: pointer.Bool(true),      // overridden
+		BoolPtr3: pointer.Bool(true),      // overridden (d=nil)
+		BoolPtr4: pointer.Bool(true),      // Not overridden (o=nil)
+		IntPtr:   pointer.Int(43),         // overridden
+		IntPtr2:  pointer.Int(200),        // Not overridden (o=nil)
+		IntPtr3:  pointer.Int(300),        // overridden (d=nil)
+		IntPtr4:  pointer.Int(0),          // overridden
+		StrPtr:   pointer.String("olleh"), // overridden
+		StrPtr2:  pointer.String("world"), // Not overridden (o=nil)
+		StrStrMap: map[string]string{ // merged (d, o)
+			"a": "apple",
+			"b": "beer",
+			"c": "",
+			"d": "daiquiri",
+		},
+		StrStrMap2: map[string]string{ // merged (d=nil, o)
+			"a": "america",
+			"b": "brazil",
+		},
+		StrSlice: []string{"gamma", "delta", "alpha", "beta"}, // appended (o, d)
+		StrArray: [2]string{"california", "colorado"},         // overridden
+		Struct: FooChild{
+			Int:     -43,                                                                                 // overridden
+			Int2:    0,                                                                                   // overridden (o=zero)
+			Str:     "ruojnob",                                                                           // overridden
+			Str2:    "",                                                                                  // overridden (o=empty)
+			IP:      net.ParseIP("192.168.11.1"),                                                         // overridden
+			IPSlice: []net.IP{net.ParseIP("10.0.0.3"), net.ParseIP("10.0.0.1"), net.ParseIP("10.0.0.2")}, // appended (o, d)
+			StrStrPtrMap: map[string]*string{ // merged (d, o)
+				"a": pointer.String("appricot"),
+				"b": pointer.String("bacardi"),
+				"c": pointer.String("cinzano"),
+				"d": pointer.String(""),
+			},
+		},
+	}
+
+	x, err := Merge(d, o)
+	assert.NilError(t, err)
+	logX(t, d, "d")
+	logX(t, o, "o")
+	logX(t, x, "x")
+	assert.DeepEqual(t, expected, x)
+}
+
+func logX(t testing.TB, x interface{}, format string, args ...interface{}) {
+	// Print in JSON for human readability
+	j, err := json.Marshal(x)
+	assert.NilError(t, err)
+	t.Logf(format+": %s", append(args, string(j))...)
+}
```

Comment:
`args` is never used; `format` is really just a name:

```suggestion
func logX(t testing.TB, x interface{}, name string) {
	// Print in JSON for human readability
	j, err := json.Marshal(x)
	assert.NilError(t, err)
	t.Logf("%s: %s", name, string(j))
}
```

Personally I would also swap the order of `x` and `name`, but it doesn't really matter.